File: java/scf-simple/src/test/java/org/mydotey/scf/source/stringproperty/environmentvariable/EnvironmentVariableConfigurationSourceTest.java
package org.mydotey.scf.source.stringproperty.environmentvariable;

import org.junit.Assert;
import org.junit.Test;
import org.mydotey.scf.ConfigurationManager;
import org.mydotey.scf.ConfigurationManagerConfig;
import org.mydotey.scf.Property;
import org.mydotey.scf.PropertyConfig;
import org.mydotey.scf.facade.ConfigurationManagers;
import org.mydotey.scf.facade.ConfigurationProperties;
import org.mydotey.scf.facade.StringPropertySources;

/**
 * @author koqizhao
 *
 * May 22, 2018
 */
public class EnvironmentVariableConfigurationSourceTest {

    protected ConfigurationManager createManager() {
        ConfigurationManagerConfig managerConfig = ConfigurationManagers.newConfigBuilder().setName("test")
                .addSource(1, StringPropertySources.newEnvironmentVariableSource("environment-variable")).build();
        System.out.println("manager config: " + managerConfig + "\n");
        return ConfigurationManagers.newManager(managerConfig);
    }

    @Test
    public void testGetProperties() {
        ConfigurationManager manager = createManager();
        PropertyConfig<String, String> propertyConfig = ConfigurationProperties.<String, String> newConfigBuilder()
                .setKey("not-exist").setValueType(String.class).build();
        Property<String, String> property = manager.getProperty(propertyConfig);
        System.out.println("property: " + property + "\n");
        Assert.assertEquals(null, property.getValue());

        propertyConfig = ConfigurationProperties.<String, String> newConfigBuilder().setKey("not-exist2")
                .setValueType(String.class).setDefaultValue("default").build();
        property = manager.getProperty(propertyConfig);
        System.out.println("property: " + property + "\n");
        Assert.assertEquals("default", property.getValue());

        propertyConfig = ConfigurationProperties.<String, String> newConfigBuilder().setKey("PATH")
                .setValueType(String.class).setDefaultValue("default").build();
        property = manager.getProperty(propertyConfig);
        System.out.println("property: " + property + "\n");
        Assert.assertNotNull(property.getValue());
        Assert.assertNotEquals("default", property.getValue());
    }

}

File: dotnet/test/MyDotey.SCF.SimpleTest/Source/StringProperty/EnvironmentVariable/EnvironmentVariableConfigurationSourceTest.cs
using System;

using Xunit;
using MyDotey.SCF.Facade;

namespace MyDotey.SCF.Source.StringProperty.EnvironmentVariable
{
    /**koqizhao
     * @author 
     *
     * May 22, 2018
     */
    public class EnvironmentVariableConfigurationSourceTest
    {
        protected ConfigurationManager createManager()
        {
            ConfigurationManagerConfig managerConfig = ConfigurationManagers.newConfigBuilder().setName("test")
                    .addSource(1, StringPropertySources.newEnvironmentVariableSource("environment-variable")).build();
            Console.WriteLine("manager config: " + managerConfig + "\n");
            return ConfigurationManagers.newManager(managerConfig);
        }

        [Fact]
        public void testGetProperties()
        {
            ConfigurationManager manager = createManager();
            PropertyConfig<String, String> propertyConfig = ConfigurationProperties.newConfigBuilder<String, String>()
                    .setKey("not-exist").build();
            Property<String, String> property = manager.getProperty(propertyConfig);
            Console.WriteLine("property: " + property + "\n");
            Assert.Null(property.getValue());

            propertyConfig = ConfigurationProperties.newConfigBuilder<String, String>().setKey("not-exist2")
                    .setDefaultValue("default").build();
            property = manager.getProperty(propertyConfig);
            Console.WriteLine("property: " + property + "\n");
            Assert.Equal("default", property.getValue());

            propertyConfig = ConfigurationProperties.newConfigBuilder<String, String>().setKey("PATH")
                    .setDefaultValue("default").build();
            property = manager.getProperty(propertyConfig);
            Console.WriteLine("property: " + property + "\n");
            Assert.NotNull(property.getValue());
            Assert.NotEqual("default", property.getValue());
        }
    }
}